### PR TITLE
DDCE-4359 Fix a11y issues

### PR DESCRIPTION
**Pull request by gastonracui**
# DDCE-4359

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## Checklist PR Raiser
- [ ]  I've updated the README if there are functionality changes
- [ ]  I've executed the project script to check Dependencies and update them if they're non-breaking changes
- [ ]  I've executed Scalastyle and fixed any warnings that can be resolved quickly
- [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
- [ ]  I've included appropriate tests with any code I've added (Unit, Integration), and it meets Coverage
- [ ]  I've squashed my commits - including the JIRA issue number in the commit message
- [ ]  I've contacted a BA to create a task if any of these actions would result in a larger piece of work

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -41,5 +41,5 @@ scalacOptions ++= Seq(
   "-Wconf:cat=unused-imports&src=views/.*:s"
 )
 
-addCommandAlias("scalafmtAll", "all scalafmtSbt scalafmt Test/scalafmt")
-addCommandAlias("scalastyleAll", "all scalastyle Test/scalastyle")
+addCommandAlias("scalafmtAll", "all scalafmtSbt scalafmt Test/scalafmt IntegrationTest/scalafmt A11y/scalafmt")
```
Comment (lmugnai) [inline build.sbt:44]: There are no Integration tests in this repo. It can be removed from here.

File: build.sbt [hunk 2]
```diff
@@ -41,5 +41,5 @@ scalacOptions ++= Seq(
...
+addCommandAlias("scalastyleAll", "all scalastyle Test/scalastyle IntegrationTest/scalastyle A11y/scalastyle")
```
Comment (lmugnai) [inline build.sbt:45]: There are no Integration tests in this repo. It can be removed from here.

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=1.7.2
+sbt.version=1.8.2
```
Comment (lmugnai) [inline project/build.properties:1]: Revert this back to 1.7.2 as it's the recommended version on the platform.

**Comment by lmugnai** [comment]
@gastonracui  Can you try and increase coverage a little?

**Comment by lmugnai** [comment]
@gastonracui can you update the following deps please?
[info]   uk.gov.hmrc:bootstrap-frontend-play-28  : 7.16.0         -> 7.21.0        
[info]   uk.gov.hmrc:bootstrap-test-play-28:test : 7.16.0         -> 7.21.0        
[info]   uk.gov.hmrc:play-frontend-hmrc          : 7.12.0-play-28 -> 7.16.0-play-28
[info]   uk.gov.hmrc:tax-year                    : 3.2.0          -> 3.3.0

File: test/views/helper/PBIKViewSpec.scala [hunk 4]
```diff
@@ -95,6 +95,75 @@ trait PBIKViewBehaviours extends PlaySpec with JsoupMatchers {
   }
 }
 
+trait PBIKNoViewBehaviours extends PlaySpec with JsoupMatchers {
```
Comment (EdCope) [inline test/views/helper/PBIKViewSpec.scala:98]: I don't think duplicating the behaviour trait and its methods is necessary. Rather make implicits for the view and use the same trait.

```
trait PBIKViewBehaviours extends PlaySpec with JsoupMatchers {
  def doc(implicit view: Html): Document = Jsoup.parse(view.toString())

  def pageWithTitle(titleText: String)(implicit view: Html): Unit =
    "have a static title" in {
      doc.title must include(titleText)
    }
```

**Review by EdCope — CHANGES_REQUESTED** [review]

File: test/views/helper/PBIKViewSpec.scala (hunk 4 above)
Comment (gastonracui) [reply test/views/helper/PBIKViewSpec.scala:98]: i tried with that and didnt work, it needed to have the view passed as a param

Comment (EdCope) [reply test/views/helper/PBIKViewSpec.scala:98]: I have it working, when i was trying to understand why. When you're free can look at it on yours.

File: test/views/templatemodels/CheckboxIdSpec.scala [hunk 5]
```diff
@@ -33,11 +33,11 @@ class CheckboxIdSpec extends PBIKViewSpec {
   def viewWithForm(form: Form[RegistrationList]): Html =
     nextTaxYearView(form, additive = true, taxYearRange, List(), List(), List(), List(), Some(1), EmpRef("", ""))
 
-  override def view: Html = viewWithForm(formMappings.objSelectedForm)
+//  implicit def view: Html = viewWithForm(formMappings.objSelectedForm)
```
Comment (EdCope) [inline test/views/templatemodels/CheckboxIdSpec.scala:36]: Please remove comment code

File: test/views/helper/PBIKViewSpec.scala [hunk 6]
```diff
@@ -28,71 +28,72 @@ import play.twirl.api.Html
...
       doc must haveBackLink
     }
 
-  def pageWithIdAndText(pageText: String, id: String): Unit =
+  def pageWithIdAndText(pageText: String, id: String)(implicit view: Html): Unit =
     s"have a static text ($pageText) with id ($id)" in {
       doc must haveElementWithIdAndText(pageText, id)
     }
 
   //scalastyle:off null
```
Comment (EdCope) [inline test/views/helper/PBIKViewSpec.scala:59]: null comment from before. Was probably on the deleted trait

**Review by EdCope — CHANGES_REQUESTED** [review]

**Review by EdCope — APPROVED** [review]

**Pull request merged**